pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,1095 @@
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``expand``. ``None`` if no decomposition is defined via ``expand``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    tape = wrap_op_method(instance, "expand", DecompositionUndefinedError)()
+    return qml.matrix(tape, wire_order=wires) if isinstance(tape, qml.tape.QuantumTape) else tape
+
+
+def matrix_from_single_qubit_rot_angles(op, par, wires):
+    r"""Get the matrix of an operation, using its ``single_qubit_rot_angles``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``single_qubit_rot_angles``. ``None`` if no rotation angles are
+        defined via ``single_qubit_rot_angles`` or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    try:
+        angles = instance.single_qubit_rot_angles()
+    except (AttributeError, NotImplementedError):
+        return None
+    with qml.tape.QuantumTape() as tape:
+        qml.RZ(angles[0], wires=wires)
+        qml.RY(angles[1], wires=wires)
+        qml.RZ(angles[2], wires=wires)
+    return qml.matrix(tape)
+
+
+def matrix_from_generator(op, par, wires):
+    r"""Get the matrix of an operation, using its ``generator``.
+
+    .. warning::
+
+        Only carries out the test for single-parameter gates.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``generator``. ``None`` if no generator is defined via ``generator``
+        or ``Exception`` if an error occured.
+    """
+    if len(par) != 1:
+        return None
+    instance = op(*par, wires=wires)
+    try:
+        gen = qml.generator(instance, "observable")
+    except GeneratorUndefinedError:
+        return None
+    mat = qml.matrix(gen)
+
+    return la.expm(1j * par[0] * mat)
+
+
+decomposition_methods = [
+    matrix_from_single_qubit_rot_angles,
+    matrix_from_hardcoded_matrix,
+    matrix_from_sparse_matrix,
+    matrix_from_terms,
+    matrix_from_decomposition,
+    matrix_from_generator,
+]
+
+
+class CheckerError(Exception):
+    """An internal error raised in OperationChecker used to mark specific exceptions."""
+
+
+class OperationChecker:
+    r"""Check one or multiple operation subclasses to define all required properties,
```

Comment:
```suggestion
    r"""Check one or multiple operator subclasses to define all required properties,
```

I know, we have to get used to the new language where everything should just work for operators, and "operation" will slowly looses it's obscure meaning as a "gate" :)

Reply:
Fair enough! :) So the whole thing should be called `OperatorChecker`, I suppose?